<a href="https://colab.research.google.com/github/saoter/AQM2023/blob/main/Workshop%205/Statistic_workshop_5.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Exercise for Chi-square test in statistics with python
##About the exercises
These are based on the book mainly A15.17, and A15.33 in the book on pages 504

## Exercise 15.17



In an election held last year that was contested by three parties Party A captured 31% of voes, Party B GARNERED 51% and Party C received the remaining vote. A survey on 1200 voters asked each to identify the party that they would vote for in the next election. These results were recorded where 1= Party A, 2= Party B and 3= Party C. can we infer at the 10% significance level that voter support has changes since the election?

    Load the dataset and examine its structure.
    Calculate the observed number of voters for each party from the survey.
    Based on the previous election results, calculate the expected number of voters for each party if there was no change in voter support.
    Perform the Chi-Squared Goodness-of-Fit Test to compare the observed and expected frequencies.
    Make a conclusion based on the p-value and the 10% significance level.

Now, let's start by loading the dataset and exploring its structure.

In [10]:
import pandas as pd
import numpy as np
from scipy.stats import chisquare, chi2_contingency


# Load the dataset

df = pd.read_excel('https://github.com/saoter/AQM2023/raw/main/Workshop%205/data/Xr15-17.xlsx')

# Display the first few rows of the dataframe
df.head()


Party
0      2
1      3
2      1
3      2
4      2

The dataset contains a single column labeled "Party," which records the party that each of the 1200 voters would vote for in the next election, with the parties encoded as 1, 2, and 3.

We will proceed with counting the number of voters for each party from this survey (observed frequencies). Then, using the previous election results, we will calculate the expected number of voters for each party if there was no change in voter support (expected frequencies). Finally, we'll apply the Chi-Squared Goodness-of-Fit Test.

In [8]:
# Calculate the observed frequencies of voters for each party
observed_frequencies = df['Party'].value_counts().sort_index()

# Calculate the expected frequencies based on the previous election results
# Party A: 31%, Party B: 51%, Party C: 18%
total_voters = len(df)
expected_frequencies = pd.Series({
    1: total_voters * 0.31,  # Party A
    2: total_voters * 0.51,  # Party B
    3: total_voters * 0.18   # Party C
})

# Display the observed and expected frequencies
observed_frequencies, expected_frequencies


(1    408
 2    571
 3    221
 Name: Party, dtype: int64,
 1    372.0
 2    612.0
 3    216.0
 dtype: float64)

Here are the observed and expected frequencies of voters for each party:

Observed Frequencies:

    Party A (1): 408 voters
    Party B (2): 571 voters
    Party C (3): 221 voters

Expected Frequencies (based on the last election results):

    Party A (1): 372 voters (31% of 1200 voters)
    Party B (2): 612 voters (51% of 1200 voters)
    Party C (3): 216 voters (18% of 1200 voters)

Next, we'll perform the Chi-Squared Goodness-of-Fit Test to determine whether there is a significant difference between the observed and expected frequencies at the 10% significance level. We want to test if voter support has changed since the last election.

In [9]:
# Perform chi-squared goodness-of-fit test
chi2_stat, p_val = chisquare(observed_frequencies, expected_frequencies)

chi2_stat, p_val


(6.346343734626467, 0.041870579134682)

The results of the Chi-Squared Goodness-of-Fit Test are:

    χ2χ2 statistic: ≈6.35≈6.35
    p-value: ≈0.042≈0.042

Interpretation:

We are testing at a 10% significance level (α=0.10α=0.10). Since the p-value is approximately 0.042, which is less than 0.10, we reject the null hypothesis.
Conclusion:

We have enough evidence to infer that voter support has changed since the last election at the 10% significance level.

## Exercice A15.33


Conduct all test a the 5% significance level, Use a 95% confidence level for estimates
According to the US Census there were 220958853 adults in the US in 2010
Does home ownership (HOUSCL: 1 = Own, 2= Not) affect how much is spent on food away from home (FOODAWAY)? Perform a statistical analysis to determine whether there is enough evidence to infer that middle class home owners spend less on food away than do middle call households who rent.

    Load the data and filter it to include only middle-class adults.
    Create a contingency table of spending on food away from home by home ownership status.
    Perform a Chi-squared test to determine if there is a significant association between home ownership and spending on food away from home.
    Interpret the results at a 5% significance level.

In [17]:
import pandas as pd
import numpy as np
from scipy.stats import chisquare, chi2_contingency


# Load the dataset

df_mc = pd.read_excel('/content/MC.xlsx')

# Display the first few rows of the dataframe
df_mc.head()

CASEID  Order  NETWORTH  AGE  HHSEX  RACE  EDCL  EDUC  FAMSTRUCT  HOUSECL  \
0  151255   2594  61215.26   62      1     1     2    12          5        1   
1  159500   2595  61268.85   80      2     2     4    16          3        1   
2  171430   2596  61302.07   70      1     1     4    14          4        1   
3  169990   2597  61343.87   51      1     1     4    14          5        1   
4  167275   2598  61515.34   30      1     2     2    12          1        2   

   ...  BUSSEFARMINC  INTDIVINC  SSRETINC  TRANSFOTHINC   WAGEINC   KGHOUSE  \
0  ...          0.00        0.0      0.00       8932.98  57737.52   5358.48   
1  ...          0.00        0.0  20698.36          0.00      0.00  52513.09   
2  ...          0.00        0.0  21787.74          0.00  87150.98  -4286.78   
3  ...       2396.65        0.0      0.00          0.00  43575.49  -7501.87   
4  ...          0.00        0.0      0.00          0.00  59916.30      0.00   

    KGTOTAL  FOODHOME  FOODDELV  FOODAWAY  
0   5358.48   5572.82      0.00    257.21  
1  52513.09   3215.09      0.00   1114.56  
2  -4286.78  16718.45      0.00   3900.97  
3  -7501.87   5572.82      0.00    643.02  
4      0.00   5572.82   1671.85   1671.85  

[5 rows x 57 columns]

In [23]:
df_mc.columns

Index(['CASEID', 'Order', 'NETWORTH', 'AGE', 'HHSEX', 'RACE', 'EDCL', 'EDUC',
       'FAMSTRUCT', 'HOUSECL', 'KIDS', 'MARRIED', 'OCCAT1', 'OCCAT2', 'LF',
       'INDCAT', 'EXPENSHILO', 'TURNDOWN', 'BNKRUPLAST5', 'NOCCBAL', 'ASSET',
       'NETWORTH.1', 'FIN', 'LIQ', 'MMA', 'CHECKING', 'SAVING', 'CDS',
       'STOCKS', 'CASHLI', 'HDEBT', 'DEBT', 'LATE', 'NH_MORT', 'CCBAL',
       'INSTALL', 'EDN_INST', 'VEH_INST', 'PAYMORT1', 'PAYVEH1', 'PAYEDU1',
       'REVPAY', 'MORTPAY', 'CONSPAY', 'TPAY', 'INCOME', 'NORMINC',
       'BUSSEFARMINC', 'INTDIVINC', 'SSRETINC', 'TRANSFOTHINC', 'WAGEINC',
       'KGHOUSE', 'KGTOTAL', 'FOODHOME', 'FOODDELV', 'FOODAWAY'],
      dtype='object')

The dataset contains multiple columns, including "HOUSECL" for home ownership and "FOODAWAY" for spending on food away from home. We will focus on these two columns for our analysis.

Given that the exercise is related to the Chi-squared test, we'll first categorize "FOODAWAY" spending into different bins and then create a contingency table to compare the spending distribution between homeowners and renters. We'll perform the Chi-squared test on this contingency table.

Let's proceed with preparing the data for the Chi-squared test.

In [19]:
# Filter the dataframe to include only relevant columns and create a copy
df_filtered = df_mc[['HOUSECL', 'FOODAWAY']].copy()

# Bin the FOODAWAY spending into categories for chi-squared test
bins = [0, 1000, 2000, 3000, 4000, 5000, np.inf]  # Change this as needed
labels = ['0-1000', '1001-2000', '2001-3000', '3001-4000', '4001-5000', '5000+']
df_filtered['FOODAWAY_BIN'] = pd.cut(df_filtered['FOODAWAY'], bins=bins, labels=labels, right=False)


We have created a contingency table showing the distribution of spending on food away from home, categorized into different bins, for homeowners (HOUSECL = 1) and renters (HOUSECL = 2).

In [21]:
# Create a contingency table
contingency_table = pd.crosstab(df_filtered['HOUSECL'], df_filtered['FOODAWAY_BIN'])

# Display the contingency table
contingency_table

FOODAWAY_BIN  0-1000  1001-2000  2001-3000  3001-4000  4001-5000  5000+
HOUSECL                                                                
1                425        290        193         74         24     94
2                 63         49         33         23          4     24




Now we will perform the Chi-squared test on this contingency table to determine if there is a significant association between home ownership and the amount spent on food away from home. We are conducting this test at a 5% significance level.

In [22]:
# Perform chi-squared test on the contingency table
chi2_stat, p_val, dof, expected = chi2_contingency(contingency_table)

chi2_stat, p_val, dof, expected


(10.118160292232368,
 0.07195562536947712,
 5,
 array([[414.19753086, 287.73148148, 191.82098765,  82.33024691,
          23.7654321 , 100.15432099],
        [ 73.80246914,  51.26851852,  34.17901235,  14.66975309,
           4.2345679 ,  17.84567901]]))

The results of the Chi-squared test are:

    χ2χ2 statistic: ≈10.12≈10.12
    p-value: ≈0.072≈0.072
    Degrees of freedom: 5

Interpretation:

We are testing at a 5% significance level (α=0.05α=0.05). Since the p-value is approximately 0.072, which is greater than 0.05, we fail to reject the null hypothesis.
Conclusion:

There isn't enough evidence at the 5% significance level to conclude that home ownership affects how much is spent on food away from home for middle-class households, according to this Chi-squared test.